aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1042 @@
+            network_public_key: dora_node_info.network_public_key,
+            elgamal_pub_key: dora_node_info.elgamal_pub_key,
+            network_port: dora_node_info.network_port,
+            rpc_port: dora_node_info.rpc_port,
+        }
+    }
+
+    #[view]
+    /// Get the committee's id for a single node
+    public fun get_committee_id_for_node(
+        com_store_addr: address,
+        node_address: address
+    ): u64 acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let id = *simple_map::borrow(&committee_store.node_to_committee_map, &node_address);
+        id
+    }
+
+    #[view]
+    /// Get the dora node peers vector for a single node
+    public fun get_peers_for_node(
+        com_store_addr: address,
+        node_address: address
+    ): vector<NodeData> acquires CommitteeInfoStore {
+        let committee_id = get_committee_id_for_node(com_store_addr, node_address);
+        let this_node = get_node_info(com_store_addr, committee_id, node_address);
+        let dora_node_info = get_committee_info(com_store_addr, committee_id);
+        let (_, index) = vector::index_of(&dora_node_info, &this_node);
+        vector::remove(&mut dora_node_info, index);
+        dora_node_info
+    }
+
+    #[view]
+    /// Check if the committee has a valid dkg
+    public fun does_com_have_dkg(com_store_addr: address, com_id: u64): bool acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow(&committee_store.committee_map, &com_id);
+        committee.has_valid_dkg
+    }
+
+    /// Update the dkg flag
+    public fun update_dkg_flag(
+        com_store_addr: address,
+        owner_signer: &signer,
+        com_id: u64,
+        flag_value: bool
+    ) acquires CommitteeInfoStore {
+        // Only the OwnerCap capability can access it
+        let _acquire = &capability::acquire(owner_signer, &OwnerCap {});
+        let committee_store = borrow_global_mut<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow_mut(&mut committee_store.committee_map, &com_id);
+        committee.has_valid_dkg = flag_value;
+    }
+
+    /// This function is used to add a new committee to the store
+    public entry fun upsert_committee(
+        com_store_addr: address,
+        owner_signer: &signer,
+        id: u64,
+        node_addresses: vector<address>,
+        ip_public_address: vector<vector<u8>>,
+        dora_public_key: vector<vector<u8>>,
+        network_public_key: vector<vector<u8>>,
+        elgamal_pub_key: vector<vector<u8>>,
+        network_port: vector<u16>,
+        rpc_port: vector<u16>
+    ) acquires CommitteeInfoStore, SupraCommitteeEventHandler {
+        // Assert the length of the vector for two are the same
+        assert!(
+            vector::length(&node_addresses) == vector::length(&ip_public_address),
```

Comment:
In this function, we are using a `vector::length(&node_addresses)` more than once (6 times), to avoid it we can store it in a variable and use it everywhere.

Reply:
@nizam-supraoracles Thanks, addressed in new commit.